template/picture-apload-rds.yml

```
@@ -0,0 +1,108 @@
+AWSTemplateFormatVersion: "2010-09-09"
+Description: "RDS"
+
+Parameters:
+  RandomName:
+    Type: String
+  # VPCのRandomNameで記入した文字を記入する
+  VPCName:
+    Type: String
+
+Resources:
+    RDSDBInstance:
+        Type: "AWS::RDS::DBInstance"
+        DeletionPolicy: Delete
+        UpdateReplacePolicy: Delete
+        Properties:
+            DBInstanceIdentifier: !Sub ${RandomName}-rds
+            AllocatedStorage: "20"
+            DBInstanceClass: "db.t3.micro"
+            Engine: "mysql"
+            MasterUsername: "admin"
+            MasterUserPassword: '{{resolve:ssm-secure:picture-apload:1}}'
+            PreferredBackupWindow: "08:19-08:49"
+            BackupRetentionPeriod: 1
+            AvailabilityZone: !Sub "${AWS::Region}b"
+            PreferredMaintenanceWindow: "mon:03:52-mon:04:22"
+            MultiAZ: false
+            EngineVersion: "8.0.33"
+            AutoMinorVersionUpgrade: true
+            LicenseModel: "general-public-license"
+            PubliclyAccessible: false
+            StorageType: "gp2"
+            Port: "3306"
+            StorageEncrypted: true
+            KmsKeyId: !Sub "arn:${AWS::Partition}:kms:${AWS::Region}:${AWS::AccountId}:key/623421e8-f259-4d7f-9b9b-f8bc1921d999"
+            CopyTagsToSnapshot: true
+            MonitoringInterval: 0
+            EnableIAMDatabaseAuthentication: false
+            EnablePerformanceInsights: false
+            DeletionProtection: false
+            DBSubnetGroupName: !Ref RDSDBSubnetGroup
+            VPCSecurityGroups: 
+              - !Ref EC2SecurityGroup2
+              - !Ref EC2SecurityGroup

```

```suggestion
              - !Ref EC2SecurityGroup2
```
RDSのセキュリティグループはこちらだけでは？

---

おっしゃる通りですが、上記で回答したようにRDSのセキュリティーグループのインバウンドはEC2からの送信のみを許可するためにSourceSecurityGroupIdに送信元であるEC2を指定しなければならないが、VPC→RDS→EC2の順番で作成しているのでEC2のセキュリティーグループを指定できないのを防ぐためにRDSのテンプレートにまとめて書いているからです。

---

https://github.com/Hidetaka-Konishi/Raise_kadai_10/pull/1#discussion_r1348203982
こちらと同じ内容かと思いますので個別のコメントは省略します！